Functions/Quantification/xASL_quant_External.m

```diff
@@ -0,0 +1,706 @@
+					% For a single-TE, we have to repeat it for each volume
+					for iTE = 1:length(TEs) %So for each volume, we print a TE value
+						fprintf(FIDoptionFile, '--te%d=%.3f\n', iTE, TEs(1));
+					end
+				else
+					% For multi-TE, we print all of them
+					for iTE = 1:length(TEs) %So for each volume, we print a TE value
+						fprintf(FIDoptionFile, '--te%d=%.3f\n', iTE, TEs(iTE));
+					end
+				end
+
+				% Future extension - specify the repetitions explicitly
+				%fprintf(FIDoptionFile, '--repeats=%i\n', size(PWI, 4)/PLDAmount);
+				%fprintf(FIDoptionFile, '--repeats=1\n');
+			case 'vaby'
+				% Printing the values in the commandline (PLD=plds, LD=taus)
+				for iPLD = 1:length(PLDs)
+					if iPLD == 1
+						ExternalOptions = [ExternalOptions ' --plds=' sprintf('%.2f', PLDs(iPLD))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.2f', PLDs(iPLD))];
+					end
+				end
+
+				for iTE = 1:length(TEs) %So for each volume, we print a TE value
+					if iTE == 1
+						ExternalOptions = [ExternalOptions ' --tes=' sprintf('%.3f', TEs(iTE))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.3f', TEs(iTE))];
+					end
+				end
+			case 'basil'
+				% Specify that we run the PCASL/CASL model
+				fprintf(FIDoptionFile, '--casl\n');
+				fprintf('BASIL: (P)CASL model\n');
+
+				% For BASIL, PLDs are specified
+				if bQuantifyMultiPLD
+					for iPLD = 1:length(PLDs)
+						fprintf(FIDoptionFile, '--pld%d=%.2f\n', iPLD, PLDs(iPLD));
+					end
+				else
+					fprintf(FIDoptionFile, '--pld=%.2f\n', PLDs(1));
+				end
+		end
+
+		switch (lower(strQuantificationType))
+			case {'fabber','basil'}
+				% Print labeling durations
+				if bQuantifyMultiPLD
+					for iLabDurs = 1:length(LabDurs)
+						fprintf(FIDoptionFile, '--tau%d=%.2f\n', iLabDurs, LabDurs(iLabDurs));
+					end
+				else
+					fprintf(FIDoptionFile, '--tau=%.2f\n', LabDurs(1));
+				end
+			case 'vaby'
+				for iLD = 1:length(LabDurs) 
+					if iLD == 1
+						ExternalOptions = [ExternalOptions ' --taus=' sprintf('%.2f', LabDurs(iLD))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.2f', LabDurs(iLD))];
+					end
+				end
+		end
+end
+
+if ~isempty(regexpi(strQuantificationType, 'basil', 'once'))
+	% Act as if we do not have repeats
+	%fprintf(FIDoptionFile, '--repeats=%i\n', size(PWI, 4)/PLDAmount);
+	fprintf(FIDoptionFile, '--repeats=1\n');
+
+	% Slice-timing
+	fprintf(FIDoptionFile, '--slicedt=%f\n', x.Q.BasilSliceReadoutTime/1000);
+
+	if isfield(x.Q,'LookLocker') && x.Q.LookLocker
+		if isfield(x.Q,'FlipAngle')
+			if length(unique(x.Q.FlipAngle))>1
+				warning('Look-Locker quantification with multiple flip angles, e.g. QUASAR, is not implemented yet');
+			end
+			fprintf(option_file, '--FA=%f\n', x.Q.FlipAngle(1));
+			fprintf('BASIL: Flip angle for Look-Locker readout: %f\n', x.Q.FlipAngle(1));
+		else
+			warning('BASIL: Unknown flip angle for Look-Locker\n');
+		end
+	end
+end
+
```

Comment:
In section 4 & 5, we only have BASIL fitting parameters. Do we have that many more fitting parameters for BASIL than for FABBER/VABY? We are those for FABBER/VABY in the big if above, and for BASIL not in the same if but separate below?

Reply:
Yes. We have all those for BASIL, but not for FABBER/VABY.